go/cli/mcap/cmd/doctor.go

```diff
@@ -213,15 +213,24 @@ func (doctor *mcapDoctor) Examine() {
 	}
 
 	var lastMessageTime uint64
+	var lastToken mcap.TokenType
+	var dataEnd *mcap.DataEnd
+	var footer *mcap.Footer
 	msg := make([]byte, 1024)
 	for {
 		tokenType, data, err := lexer.Next(msg)
 		if err != nil {
 			if errors.Is(err, io.EOF) {
+				if dataEnd == nil {
+					doctor.error("File does not contain a DataEnd record (last record was %s)", lastToken.String())
+				} else if footer == nil {
```

Comment:
```suggestion
				}
				if footer == nil {
```

Reply:
Ideally there would be two errors if the file did not contain either a DataEnd or a Footer